geco/mips/knapsack.py

```diff
@@ -4,31 +4,36 @@
 import pyscipopt as scip
 
 
-def knapsack(n, seed=0):
+def yang(n, seed):
     """
-    This function is based on the MIP generation techniques described in
-    Yu Yang, Natashia Boland, Bistra Dilkina, Martin Savelsbergh,
-    "Learning Generalized Strong Branching for Set Covering,
-    Set Packing, and 0-1 Knapsack Problems", 2020.
+    Generates knapsack instance parameters according to:
+        Yu Yang, Natashia Boland, Bistra Dilkina, Martin Savelsbergh,
+        "Learning Generalized Strong Branching for Set Covering,
+        Set Packing, and 0-1 Knapsack Problems", 2020.
     """
     random.seed(seed)
-    model = scip.Model("Yang Knapsack")
 
     def draw_value():
         return random.randint(1, 10 * n)
 
-    # add variables and their cost
-    vars = []
-    for i in range(n):
-        profit = draw_value()
-        var = model.addVar(lb=0, ub=1, obj=profit, vtype="B")
-        vars.append(var)
+    profit = [draw_value() for _ in range(n)]
+    weights = [draw_value() for _ in range(n)]
+    capacity = math.floor(sum(weights) / 5)
+
+    return profit, weights, capacity
+
 
+def knapsack(weights, profits, capacity, name="Knapsack"):
```

Comment:
What about naming this yang and the other one yang_params?, so that we could add another knapsack generator to this file without confusion.

Reply:
I think the general name here is better. Every default definition of knapsack consists of exactly those parameters (see e.g. [wiki](https://en.wikipedia.org/wiki/Knapsack_problem))